### PR TITLE
MGMT-6768: Add NetworkType to KubeAPI

### PR DESCRIPTION
**Pull request by slaviered**
# Assisted Pull Request

## Description

NetworkType parameter is added to the Networking section in AgentClusterInstall CRD
It can have 2 possible values: OVNKubernetes or OpenShiftSDN
Validation are run in the BE and their outcome is reflect automatically in the status field

## List all the issues related to this PR

- [x] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [] None

## How was this code tested?

- [x] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [] No tests needed

## Assignees
/assign @oshercc 
/assign @ori-amizur 

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: api/hiveextension/v1beta1/agentclusterinstall_types.go [hunk 1]
```diff
@@ -210,6 +210,13 @@ type Networking struct {
 	// +kubebuilder:validation:MaxItems=1
 	// +optional
 	ServiceNetwork []string `json:"serviceNetwork,omitempty"`
+
+	//NetworkType is the Container Network Interface (CNI) plug-in to install
+	//The default value is OpenShiftSDN for IPv4 and OVNKubernetes for IPv6
```
Comment (mhrivnak) [inline api/hiveextension/v1beta1/agentclusterinstall_types.go:215]: How and when is the default determined and applied?

Comment (slaviered) [reply api/hiveextension/v1beta1/agentclusterinstall_types.go:215]: I reinstated the function that determine the default before reconciliation

File: subsystem/kubeapi_test.go [hunk 2]
```diff
@@ -2193,6 +2284,10 @@ var _ = Describe("[kube-api]cluster installation", func() {
 		checkAgentCondition(ctx, day2Host2.ID.String(), v1beta1.ValidatedCondition, v1beta1.ValidationsPassingReason)
 	})
 
+	It("deploy clusterDeployment with networkType configuration", func() {
+
+	})
```
Comment (carbonin) [inline subsystem/kubeapi_test.go:2289]: @slaviered did you mean to add some test content here?

Comment (slaviered) [reply subsystem/kubeapi_test.go:2289]: No. I will remove this

**Pull request merged**